frontend/src/components/BikeGallery/BikeGallery.tsx

```diff
@@ -1,21 +1,34 @@
 import React from 'react'
-import { Bike } from '../../models/Bike'
 import BikeCard from '../BikeCard/BikeCard'
 import './BikeGallery.css'
+import useAuth from '../../hooks/useAuth'
+import {useNavigate} from 'react-router-dom'
+import AddBike from "../AddBike/AddBike";
+import useBikesApi from "../../hooks/useBikesApi";
+import Layout from "../Layout/Layout";
 
 type BikeGalleryProps = {
-   bikes: Bike[]
-   editBike: (updatedBike: Bike) => void
-   deleteBike: (id: string) => void
+   addBikeInputRef: React.MutableRefObject<HTMLInputElement>
 }
 
 function BikeGallery(props: BikeGalleryProps) {
+   const currentUser = useAuth(false)
+    const { bikes, addBike, editBike, deleteBike, loading } = useBikesApi()
+    const navigate = useNavigate()
+    !currentUser && navigate("/login")
```

Comment:
Instead of this line, you could write `useAuth(true)` in line 15 we think :)

Reply:
👍